packages/pwa-buildpack/src/WebpackTools/plugins/MagentoRootComponentsPlugin/index.js

```diff
@@ -46,21 +47,37 @@ class MagentoRootComponentsPlugin {
                     // us to grab the ID during the emit phase
                     moduleByPath.set(mod.resource, mod);
                 }
+
                 // To create a unique chunk for each RootComponent, we want to inject
                 // a dynamic import() for each RootComponent, within each entry point.
-                const isAnEntry = compilation.entries.some(entryMod => {
-                    // Check if the module being constructed matches a defined entry point
-                    if (mod === entryMod) return true;
-                    if (!entryMod.identifier().startsWith('multi')) {
-                        return false;
-                    }
 
-                    // If a multi-module entry is used (webpack-dev-server creates one), we
-                    // need to try and match against each dependency in the multi module
-                    return entryMod.dependencies.some(
-                        singleDep => singleDep.module === mod
-                    );
-                });
+                // But identifying entry points is hard!
+
+                // Top-level modules injected by a downstream "issuer" are not
+                // entry points.
+                let isEntrySimpleTest = mod => !mod.issuer;
+                if (this.phase === 'development') {
+                    // Unless we're in development mode. In that case,
+                    // the best we can do is test if the entry is in node_modules.
+                    isEntrySimpleTest = mod =>
+                        mod.resource.indexOf('/node_modules/') !== -1;
+                }
```

Comment:
Super minor quibble:

```js
const isEntrySimpleTest = this.phase === 'development'
    ? mod => mod.resource.includes('/node_modules/')
    : mod => !mod.issuer;
```

Reply:
Thanks, I think I'd been staring at that too long.